src/ZoneServer/Skills/Handlers/Highlander/SkyLiner.cs

```diff
@@ -0,0 +1,130 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Melia.Shared.Data.Database;
+using Melia.Shared.L10N;
+using Melia.Shared.Tos.Const;
+using Melia.Shared.World;
+using Melia.Zone.Network;
+using Melia.Zone.Skills.Combat;
+using Melia.Zone.Skills.Handlers.Base;
+using Melia.Zone.Skills.SplashAreas;
+using Melia.Zone.World.Actors;
+using Melia.Zone.World.Actors.Characters;
+using Melia.Zone.World.Actors.CombatEntities.Components;
+using static Melia.Zone.Skills.SkillUseFunctions;
+
+namespace Melia.Zone.Skills.Handlers.Highlander
+{
+	/// <summary>
+	/// Handler for the Highlander skill SkyLiner.
+	/// </summary>
+	[SkillHandler(SkillId.Highlander_SkyLiner)]
+	public class SkyLiner : IGroundSkillHandler, IDynamicCasted
+	{
+		private const int TotalHits = 4;
+		private bool keepAttacking = false;
```

Comment:
This is a good catch.  I've rewritten it to use the Character variables.  This does force a cast to Character (since ICombatEntity doesn't have variables), so if this skill is used by any other type of entity, they can't repeat it.  I'm not sure if this is relevant.